mediawiki.go

```diff
@@ -139,7 +139,7 @@ func checkError(response []byte) error {
 	var mwerror mwError
 	err := json.Unmarshal(response, &mwerror)
 	if err != nil {
-		return nil
```

Comment:
This is a little bit unclear, but the reason we return a nil instead of the error here is that we are trying to determine if mediawiki has sent an error to us. The somewhat cheaty way I did this is to attempt to unmarshal their response in to the mwerror struct, if it works: it is an error, if it doesn't work, it probably isn't an error.

Reply:
well why not ignore the unmarshal err all together? if we are not going to use it then better just move on to the mwerror check.

Reply:
at the moment if we have an unmarshal error it would still return nil as if the mwerror is also nil.

Reply:
This is the goal, if we fail to unmarshal the mwerror (since it might not be an error, it could just be a regular response) we want to return "nil", i.e. it is not a mwerror.